src/query/service/src/servers/http/clickhouse_handler.rs

```diff
@@ -290,6 +290,52 @@ pub async fn clickhouse_handler_post(
                 "clickhouse insert with format {:?}, value {}",
                 input_context, *start
             );
+            let compression_alg = input_context.get_compression_alg("").map_err(BadRequest)?;
+            let start = *start;
+            handle = Some(ctx.spawn(async move {
+                gen_batches(
+                    sql,
+                    start,
+                    input_context.read_batch_size,
+                    tx,
+                    compression_alg,
+                )
+                .await
+            }));
+        } else if let InsertInputSource::StreamingWithFileFormat(
+            option_settings,
+            start,
+            input_context_ref,
+        ) = &mut insert.source
+        {
+            let (tx, rx) = tokio::sync::mpsc::channel(2);
+            let to_table = ctx
+                .get_table(&insert.catalog, &insert.database, &insert.table)
+                .await
+                .map_err(InternalServerError)?;
+
+            // override settings
+            settings
+                .set_file_format_options(option_settings)
+                .map_err(InternalServerError)?;
+
+            let input_context = Arc::new(
+                InputContext::try_create_from_insert(
```

Comment:
we need a new `try_create_from_insert_v2` which do not use settings to pass options.

Reply:
you can refer to try_create_from_copy

Reply:
good idea, with `try_create_from_insert_v2` we can avoid use settings.